fe/fe-core/src/main/java/com/starrocks/clone/TabletScheduler.java

```diff
@@ -1185,7 +1185,13 @@ private boolean deleteLocationMismatchReplica(TabletSchedCtx tabletCtx, boolean
                 if (backend != null) {
                     Pair<String, String> location = backend.getSingleLevelLocationKV();
                     if (location != null && matchedLocations.contains(location)) {
-                        dupReplica = replica;
```

Comment:
how about iterating replicas backwards and remove the replica with duplicate location?

Reply:
current implement doesn't guarantee that the source replica must be deleted via `deleteReplicaChosenByRebalancer` after balancing.

Reply:
Hi @wyb I have adopted your suggestion and tested it in the production environment. Both approaches work well.